path.go

```diff
@@ -55,6 +76,133 @@ func NewPathFromSVGString(svg string) (*Path, error) {
+	p.path = skia.PathNew()
+
+	for _, op := range svgPath.Path {
+		switch op := op.(type) {
+		case svg.OpMoveTo:
+			p.MoveTo(float32(op.X)/64.0, float32(op.Y)/64.0)
+		case svg.OpLineTo:
+			p.LineTo(float32(op.X)/64.0, float32(op.Y)/64.0)
+		case svg.OpQuadTo:
+			p.QuadTo(
+				float32(op[0].X)/64.0, float32(op[0].Y)/64.0,
+				float32(op[1].X)/64.0, float32(op[1].Y)/64.0,
+			)
+		case svg.OpCubicTo:
+			p.CubicTo(
+				float32(op[0].X)/64.0, float32(op[0].Y)/64.0,
+				float32(op[1].X)/64.0, float32(op[1].Y)/64.0,
+				float32(op[2].X)/64.0, float32(op[2].Y)/64.0,
+			)
+		case svg.OpClose:
+			p.Close()
+		}
+	}
+
+	if svgPath.Style.Transform != svg.Identity {
+		p.Transform(
+			geom.Matrix[float32]{
+				ScaleX: float32(svgPath.Style.Transform.A),
+				SkewX:  float32(svgPath.Style.Transform.C),
+				TransX: float32(svgPath.Style.Transform.E),
+				SkewY:  float32(svgPath.Style.Transform.B),
+				ScaleY: float32(svgPath.Style.Transform.D),
+				TransY: float32(svgPath.Style.Transform.F),
+			})
+	}
+}
+
+func (p *Path) createFillPaint(svgPath svg.SvgPath) error {
+	if svgPath.Style.FillerColor == nil ||
+		svgPath.Style.FillOpacity == 0 {
+		return nil
+	}
+
+	p.fillPaint = NewPaint()
+	p.fillPaint.SetStyle(paintstyle.Fill)
+
+	if color, ok := svgPath.Style.FillerColor.(svg.PlainColor); ok {
+		alpha := float32(color.A) / 0xFF
+		alpha *= float32(svgPath.Style.FillOpacity)
+		p.fillPaint.SetColor(ARGB(alpha, int(color.R), int(color.G), int(color.B)))
+	} else {
+		return fmt.Errorf("unsupported path fill style %T", svgPath.Style.FillerColor)
+	}
+
+	return nil
+}
+
+func (p *Path) createStrokePaint(svgPath svg.SvgPath) error {
+	if svgPath.Style.LinerColor == nil ||
+		svgPath.Style.LineOpacity == 0 ||
+		svgPath.Style.LineWidth == 0 {
+		return nil
+	}
+
+	strokeCap, ok := strokeCaps[svgPath.Style.Join.TrailLineCap]
+	if !ok {
+		return fmt.Errorf("unsupported path stroke cap %s", svgPath.Style.Join.TrailLineCap)
+	}
+	strokeJoin, ok := strokeJoins[svgPath.Style.Join.LineJoin]
+	if !ok {
+		return fmt.Errorf("unsupported path stroke join %s", svgPath.Style.Join.LineJoin)
+	}
+
+	p.strokePaint = NewPaint()
+	p.strokePaint.SetStrokeCap(strokeCap)
+	p.strokePaint.SetStrokeJoin(strokeJoin)
+	p.strokePaint.SetStrokeWidth(float32(svgPath.Style.LineWidth))
+	p.strokePaint.SetStyle(paintstyle.Stroke)
+
+	if color, ok := svgPath.Style.LinerColor.(svg.PlainColor); ok {
+		alpha := float32(color.A) / 0xFF
+		alpha *= float32(svgPath.Style.LineOpacity)
+		p.strokePaint.SetColor(ARGB(alpha, int(color.R), int(color.G), int(color.B)))
+	} else {
+		return fmt.Errorf("unsupported path stroke style %T", svgPath.Style.LinerColor)
+	}
+
+	return nil
+}
+
+func (p *Path) draw(canvas *Canvas) {
+	if p.fillPaint != nil {
+		p.drawWithPaint(canvas, p.fillPaint)
+	}
+	if p.strokePaint != nil {
+		p.drawWithPaint(canvas, p.strokePaint)
+	}
+}
+
+func (p *Path) drawWithPaint(canvas *Canvas, paint *Paint) {
+	canvas.DrawPath(p, paint)
+}
+
```

Comment:
After thinking about it a bit, I'm inclined to think that all of these changes really should be moved over to the svg.go file, since they are all very specific to svg's and not generally applicable to the way paths can be used by clients.

I'm willing to do this rearranging/cleanup for you since I'm the one asking for what might seem like arbitrary changes. Let me know if you'd like me to do so.

Reply:
This makes sense. I was already on the fence about where the path drawing belonged.

I moved it in to `DrawableSVG#DrawInRect`.